### PR TITLE
Feature/mc 11572 configmaps

### PR DESCRIPTION
**Pull request by jasondias9**
#### Changes made
- Added doc for config maps for kubernetes plugin
#### Related PRs
- [Kubernetes SDK](https://github.com/cloudops/cloudmc-kubernetes-sdk/pull/68)

### CODE REVIEW

**Comment by jasondias9** [comment]
@m-orfanos could you also move this one to azure & gcp when you're done with the secrets api doc? 🙏

File: source/includes/kubernetes/_k8_configmaps.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+<code>GET /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/configmaps</code>
+
+Retrieve a list of all configmaps in a given [environment](#administration-environments).
+
+| Attributes                                 | &nbsp;                                                  |
+| ------------------------------------------ | ------------------------------------------------------- |
+| `id` <br/>_string_                         | The id of the configmap                                 |
+| `apiVersion` <br/>_string_                 | The API version used to retrieve this configmap         |
+| `kind` <br/>_string_                       | The type of the returned resource: ConfigMap            |
```
Comment (Gerudhoh) [inline source/includes/kubernetes/_k8_configmaps.md]: Could we change this to "The type of the returned resource. In this case, a ConfigMap." or "The type of the returned resource. ie, a ConfigMap." or "The type of the returned resource (a ConfigMap)"

File: source/includes/kubernetes/_k8_configmaps.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+```
+
+<code>GET /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/configmaps</code>
+
+Retrieve a list of all configmaps in a given [environment](#administration-environments).
+
+| Attributes                                 | &nbsp;                                                  |
+| ------------------------------------------ | ------------------------------------------------------- |
+| `id` <br/>_string_                         | The id of the configmap                                 |
+| `apiVersion` <br/>_string_                 | The API version used to retrieve this configmap         |
```
Comment (Gerudhoh) [inline source/includes/kubernetes/_k8_configmaps.md]: Also why is is configmap in some places (all lowercase) and ConfigMap in others?

File: source/includes/kubernetes/_k8_configmaps.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+| `kind` <br/>_string_                       | The type of the returned resource: ConfigMap            |
+| `metadata` <br/>_object_                   | The metadata of the configmap                           |
+| `metadata.creationTimestamp` <br/>_string_ | The date of creation of the configmap as a string       |
+| `metadata.annotations` <br/>_map_          | The annotations associated to the configmap             |
+| `metadata.name` <br/>_string_              | The name of the configmap                               |
+| `metadata.namespace` <br/>_string_         | The namespace in which the configmap is created         |
+| `metadata.uid` <br/>_object_               | The UUID of the configmap                               |
+| `metadata.selfLink` <br/>_object_          | A link uniquely identifying this config map             |
```
Comment (Gerudhoh) [inline source/includes/kubernetes/_k8_configmaps.md]: Could we change this to "A link that uniquely identified each config map" (Or you could say "this" instead of each, whatever you prefer)

File: source/includes/kubernetes/_k8_configmaps.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+| `metadata.resourceVersion` <br/>_object_   | The version returned of the configmap                   |
```
Comment (Gerudhoh) [inline source/includes/kubernetes/_k8_configmaps.md]: Can we reword this as well? What do you mean by "returned", and why did you use that word? I would suggest removing that word "returned" for two reasons: 1. Isn't it obvious that it's returned? 2: it makes the sentence flow weirdly. Could we instead say "The resource version of the config map" or something similar to that?

**Review by Gerudhoh — CHANGES_REQUESTED** [review]
Please refer to my comments regarding sentence rewording. I want to achieve as much clarity as possible in these docs.

File: source/includes/kubernetes/_k8_configmaps.md (hunk 3 above)
Comment (jasondias9) [reply source/includes/kubernetes/_k8_configmaps.md]: How about `A link that uniquely identifies this config map` ?

File: source/includes/kubernetes/_k8_configmaps.md (hunk 1 above)
Comment (jasondias9) [reply source/includes/kubernetes/_k8_configmaps.md]: So for the the type in this I kept it as `ConfigMap` since this is exact casing of the the value attributed to the `type` key in the response. For the others I wrote them as `config map` instead of `configmap` to avoid any confusion

File: source/includes/kubernetes/_k8_configmaps.md (hunk 2 above)
Comment (jasondias9) [reply source/includes/kubernetes/_k8_configmaps.md]: So for the the type in this I kept it as `ConfigMap` since this is exact casing of the the value attributed to the `type` key in the response. For the others I wrote them as `config map` instead of `configmap` to avoid any confusion [5ed38c1](https://github.com/cloudops/cloudmc-api-docs/pull/169/commits/5ed38c15409b1adf836177c22238b501dc5821c4)

File: source/includes/kubernetes/_k8_configmaps.md (hunk 1 above)
Comment (jasondias9) [reply source/includes/kubernetes/_k8_configmaps.md]: [5ed38c1](https://github.com/cloudops/cloudmc-api-docs/pull/169/commits/5ed38c15409b1adf836177c22238b501dc5821c4)

File: source/includes/kubernetes/_k8_configmaps.md (hunk 4 above)
Comment (jasondias9) [reply source/includes/kubernetes/_k8_configmaps.md]: No reason just bad wording. Updated to your suggestion: [5ed38c1](https://github.com/cloudops/cloudmc-api-docs/pull/169/commits/5ed38c15409b1adf836177c22238b501dc5821c4)

**Review by m-orfanos — APPROVED** [review]

File: source/includes/kubernetes/_k8_configmaps.md (hunk 3 above)
Comment (Gerudhoh) [reply source/includes/kubernetes/_k8_configmaps.md]: Sorry I didn't see that until now. That's great!

File: source/includes/kubernetes/_k8_configmaps.md [hunk 5]
```diff
@@ -91,17 +91,17 @@ curl -X GET \
...
-| `metadata.uid` <br/>_object_               | The UUID of the configmap                                      |
-| `metadata.selfLink` <br/>_object_          | A link uniquely identifying this config map                    |
-| `metadata.resourceVersion` <br/>_object_   | The version returned of the configmap                          |
+| Attributes                                 | &nbsp;                                                  |
+| ------------------------------------------ | ------------------------------------------------------- |
+| `id` <br/>_string_                         | The id of the config map                                |
+| `apiVersion` <br/>_string_                 | The API version used to retrieve this config map        |
+| `kind` <br/>_string_                       | The type of the returned resource. ie, ConfigMap        |
+| `metadata` <br/>_object_                   | The metadata of the config map                          |
+| `metadata.creationTimestamp` <br/>_string_ | The date of creation of the configmap as a string       |
```
Comment (Gerudhoh) [inline source/includes/kubernetes/_k8_configmaps.md]: Looks like this config map needs a space #LeaveRoomForJesus

**Review by Gerudhoh — APPROVED** [review]
LGTM! Thanks for making those changes :)

File: source/includes/kubernetes/_k8_configmaps.md (hunk 5 above)
Comment (jasondias9) [reply source/includes/kubernetes/_k8_configmaps.md]: [f17f2a6](https://github.com/cloudops/cloudmc-api-docs/pull/169/commits/f17f2a61c479aca8ff6b7d2313097cc8e72c4a95)
